sale_procurement_group_by_line/tests/test_sale_procurement_group_by_line.py

```diff
@@ -0,0 +1,144 @@
+        # Create Products
+        cls.new_product1 = cls._create_product("test_product1")
+        cls.new_product2 = cls._create_product("test_product2")
+        cls.sale = cls._create_sale_order()
+
+    @classmethod
+    def _create_product_category(cls):
+        product_ctg = cls.product_ctg_model.create({"name": "test_product_ctg"})
+        return product_ctg
+
+    @classmethod
+    def _create_product(cls, name):
+        product = cls.product_model.create(
+            {"name": name, "categ_id": cls.product_ctg.id, "is_storable": True}
+        )
+        return product
+
+    @classmethod
+    def _create_sale_order(cls):
+        """Create a Sale Order."""
+        cls.sale = cls.sale_model.create(
+            {
+                "partner_id": cls.customer.id,
+                "warehouse_id": cls.warehouse_id.id,
+                "picking_policy": "direct",
+            }
+        )
+        cls.line1 = cls.order_line_model.create(
+            {
+                "order_id": cls.sale.id,
+                "product_id": cls.new_product1.id,
+                "product_uom_qty": 10.0,
+                "name": "Sale Order Line Demo1",
+            }
+        )
+        cls.line2 = cls.order_line_model.create(
+            {
+                "order_id": cls.sale.id,
+                "product_id": cls.new_product2.id,
+                "product_uom_qty": 5.0,
+                "name": "Sale Order Line Demo2",
+            }
+        )
+        return cls.sale
+
+    def test_01_procurement_group_by_line(self):
+        self.sale.action_confirm()
+        self.assertEqual(
+            self.line2.procurement_group_id,
+            self.line1.procurement_group_id,
+            """Both Sale Order line should belong
+                         to Procurement Group""",
+        )
+        self.picking_ids = self.env["stock.picking"].search(
+            [("group_id", "in", self.line2.procurement_group_id.ids)]
+        )
+        self.picking_ids.move_ids.write({"quantity": 5})
+        wiz_act = self.picking_ids.button_validate()
+        wiz = Form(
+            self.env[wiz_act["res_model"]].with_context(**wiz_act["context"])
+        ).save()
+        wiz.process()
+        self.assertTrue(self.picking_ids, "Procurement Group should have picking")
+
+    def test_02_action_launch_procurement_rule_1(self):
+        group_id = self.proc_group_model.create(
+            {"move_type": "one", "sale_id": self.sale.id, "name": self.sale.name}
+        )
+        self.line1.procurement_group_id = group_id
+        self.line2.procurement_group_id = group_id
+        self.sale.action_confirm()
+        self.assertEqual(self.sale.state, "sale")
+        self.assertEqual(len(self.line1.move_ids), 1)
+        self.assertEqual(self.line1.move_ids.name, self.line1.display_name)
+        self.assertEqual(len(self.line2.move_ids), 1)
+        self.assertEqual(self.line2.move_ids.name, self.line2.display_name)
```

Comment:
```suggestion
        self.assertEqual(self.line1.move_ids.name, self.line1.product_id.display_name)
        self.assertEqual(len(self.line2.move_ids), 1)
        self.assertEqual(self.line2.move_ids.name, self.line2.product_id.display_name)
```

It seems it's due to https://github.com/odoo/odoo/commit/4cc2b3bcca613dfb37bb77e5fef6de6e044e3399